src/test/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaSubmissionSelectionServiceTest.java

```diff
@@ -10,72 +10,127 @@
+                jsonPath("$.submissionIds").isArray());
+        var submission = athenaSubmissionSelectionService.getProposedSubmissionId(programmingExercise, List.of(programmingSubmission1.getId()));
+        assertThat(submission).contains(programmingSubmission1.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
     void testNoSubmissionSelectionFromOne() {
-        athenaRequestMockProvider.mockSelectSubmissionsAndExpect(-1, jsonPath("$.exercise.id").value(textExercise.getId()), jsonPath("$.submissionIds").isArray());
+        athenaRequestMockProvider.mockSelectSubmissionsAndExpect("text", -1, jsonPath("$.exercise.id").value(textExercise.getId()), jsonPath("$.submissionIds").isArray());
         var submission = athenaSubmissionSelectionService.getProposedSubmissionId(textExercise, List.of(textSubmission1.getId()));
         assertThat(submission).isEmpty();
     }
 
     @Test
-    void testSubmissionSelectionFromTwo() {
-        athenaRequestMockProvider.mockSelectSubmissionsAndExpect(1, jsonPath("$.exercise.id").value(textExercise.getId()), jsonPath("$.submissionIds").isArray());
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
+    void testSubmissionSelectionFromTwoText() {
+        athenaRequestMockProvider.mockSelectSubmissionsAndExpect("text", 1, jsonPath("$.exercise.id").value(textExercise.getId()), jsonPath("$.submissionIds").isArray());
         var submission = athenaSubmissionSelectionService.getProposedSubmissionId(textExercise, List.of(textSubmission1.getId(), textSubmission2.getId()));
         assertThat(submission).contains(textSubmission1.getId());
     }
 
     @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
+    void testSubmissionSelectionFromTwoProgramming() {
+        athenaRequestMockProvider.mockSelectSubmissionsAndExpect("programming", 4, jsonPath("$.exercise.id").value(programmingExercise.getId()),
+                jsonPath("$.submissionIds").isArray());
+        var submission = athenaSubmissionSelectionService.getProposedSubmissionId(programmingExercise, List.of(programmingSubmission1.getId(), programmingSubmission2.getId()));
+        assertThat(submission).contains(programmingSubmission2.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
     void testSubmissionSelectionWithFeedbackSuggestionsDisabled() {
-        textExercise.setAssessmentType(AssessmentType.MANUAL); // disable feedback suggestions
+        textExercise.setFeedbackSuggestionsEnabled(false);
         assertThatThrownBy(() -> athenaSubmissionSelectionService.getProposedSubmissionId(textExercise, List.of(textSubmission1.getId())))
                 .isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
     void testSubmissionSelectionWithException() {
```

Comment:
Those two test cases might also need a programming equivalent, I don't know if it is really necessary though

Reply:
It felt a bit duplicated, but it doesn't hurt I guess